src/main/scala/subsystem/Configs.scala

```diff
@@ -41,6 +41,8 @@ class BaseSubsystemConfig extends Config ((site, here, up) => {
   case DebugModuleKey => Some(DefaultDebugModuleParams(site(XLen)))
   case CLINTKey => Some(CLINTParams())
   case PLICKey => Some(PLICParams())
+  case TilesLocated(InSubsystem) => 
```

Comment:
Will there be an example for what direct manipulation of the TilesLocated pattern looks like? (Not using the LegacyTileFieldHelper)

Reply:
Good suggestion, will do

Reply:
I ended up doing this example in `WithTraceGen` for the `TraceGenTile` subtype. I'm not very keen to mess too much with the existing `RocketTilesKey` `Config` alterations since it it seems hard to change them while remaining backwards compatibly with external alterations of `RocketTilesKey`. Check out `WithTraceGen` and `HeterogeneousTileExampleConfig` and let me know if that is sufficiently helpful.

Reply:
Thanks, this is pretty clear.

I noticed that manipulation of core parameters is more verbose without direct access through `CoreTilesKey`, since the type of the core must be matched, and since `AttachParams` adds another layer of case-class nesting. Given that manipulating tile-level parameters is the common use-case of the core-specific `TilesKey` s, it may be worth defining a generic macro to modify the `TileParams` matching some `desired` Tile type.

Reply:
if you can figure out a good pattern for applying a ["lens"](https://medium.com/zyseme-technology/functional-references-lens-and-other-optics-in-scala-e5f7e2fdafe) to nested case classes in Config alterations I will buy you a beer

Reply:
But yes something specific to this particular use case is probably also possible